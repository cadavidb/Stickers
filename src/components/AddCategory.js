import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState(''); // ''

    const handleInputChange = ( e ) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if ( inputValue.trim().length > 2 ) {
            setCategories( cats => [ inputValue, ...cats, ] );
            setInputValue('');
        }

    }

    return (
        <>
        <div className="background"></div>
        <div id="entrada">
        <form onSubmit={ handleSubmit }>
            <p> { inputValue } </p>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
      
        </div>
        
        <button onClick={handleSubmit} id="boton" className="animate__animated animate__shakeY">Buscar Gifs</button>
        </>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
