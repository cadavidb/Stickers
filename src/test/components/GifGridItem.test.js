import React from 'react'

import {shallow} from 'enzyme'
import {GifGridItem} from '../../components/GifGridItem'


describe('prueba 1', () => {
   
    const titulo="players";
    const url="http://www.google.com";
    const wrapper=shallow(<GifGridItem title={titulo} url={url} />)




    test('debe mostrar el componente correctamente', () => {
       
        expect(wrapper).toMatchSnapshot()
    })
    



    test('el parrafo debe tener el title', () => {
        const parrafo=wrapper.find('p');
       
        expect(parrafo.text().trim()).toBe(titulo)
    })
    

    test('debe tener la imagen , igual al url', () => {
        const urlimg=wrapper.find('img')
        expect(urlimg.props().src).toBe(url)
        
    })
    test('debe tener un animate FadeIn', () => {
        const fade=wrapper.find('div');
        expect(fade.props().className).toBe('card animate__animated animate__fadeIn')
      

        

    })
    
})
