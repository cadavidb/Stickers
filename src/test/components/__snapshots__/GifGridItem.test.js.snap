// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba 1 debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="players"
    src="http://www.google.com"
  />
  <p>
     
    players
     
  </p>
</div>
`;
